src/urllib3/connection.py

```diff
@@ -406,6 +417,12 @@ def _match_hostname(cert, asserted_hostname):
         raise
 
 
+def _get_default_user_agent():
+    from . import __version__
```

Comment:
Isn't it better to import this at the top of the module? Does python optimize repeated imports like this somehow?

Reply:
I'm not sure about performance, but it would be better to import this at the top, right. Unfortunately we failed to review this promptly and this now needs to be rebased on top of master to fix the conflicts and CI.

Reply:
Done.